src/Raven.Server/Documents/Transformers/TransformerStore.cs

```diff
@@ -115,38 +189,24 @@ private void OpenTransformers()
                         throw new AggregateException("Could not load some of the transformers", exceptions);
                 }
             }
+
+            HandleDatabaseRecordChange(null, _documentDatabase.Name);
         }
 
-        public int CreateTransformer(TransformerDefinition definition)
+        public void CreateTransformer(TransformerDefinition definition)
         {
             if (definition == null)
                 throw new ArgumentNullException(nameof(definition));
-
-            lock (_indexAndTransformerLocker)
+            
+            var index = _documentDatabase.IndexStore.GetIndex(definition.Name);
+            if (index != null)
             {
-                var index = _documentDatabase.IndexStore.GetIndex(definition.Name);
-                if (index != null)
-                {
-                    throw new IndexOrTransformerAlreadyExistException($"Tried to create a transformer with a name of {definition.Name}, but an index under the same name exist");
-                }
-                Transformer existingTransformer;
-                var lockMode = ValidateTransformerDefinition(definition.Name, out existingTransformer);
-                if (lockMode == TransformerLockMode.LockedIgnore)
-                    return existingTransformer.TransformerId;
-
-                if (existingTransformer != null)
-                {
-                    if (existingTransformer.Definition.Equals(definition))
-                        return existingTransformer.TransformerId; // no op for the same transformer
-
-                    DeleteTransformer(existingTransformer.TransformerId);
-                }
-
-                var transformerId = _transformers.GetNextIndexId();
-                var transformer = Transformer.CreateNew(transformerId, definition, _documentDatabase.Configuration.Indexing, LoggingSource.Instance.GetLogger<Transformer>(_documentDatabase.Name));
-
-                return CreateTransformerInternal(transformer, transformerId);
+                throw new IndexOrTransformerAlreadyExistException($"Tried to create a transformer with a name of {definition.Name}, but an index under the same name exist");
```

Comment:
This is always called from well known location. We are better off we handling this properly without throwing an error.